rest_framework_simplejwt/tokens.py

```diff
@@ -235,24 +235,29 @@ def blacklist(self):
 
             return BlacklistedToken.objects.get_or_create(token=token)
 
-        @classmethod
-        def for_user(cls, user):
+        def create_outstanding_token(self):
             """
-            Adds this token to the outstanding token list.
+            Mark this token as outstanding in the database.
             """
-            token = super().for_user(user)
-
-            jti = token[api_settings.JTI_CLAIM]
-            exp = token["exp"]
+            jti = self[api_settings.JTI_CLAIM]
+            exp = self["exp"]
 
             OutstandingToken.objects.create(
-                user=user,
+                user_id=self[api_settings.USER_ID_CLAIM],
```

Comment:
```suggestion
                user_pk=self[api_settings.USER_ID_CLAIM],
```

Reply:
The point is that TokenRefreshSerializer.validate() (the new caller for this helper) doesn't currently hold a User object. Using just the id here avoids needing to query the database for one.

Another alternative that comes to mind would be just doing RefreshToken.for_user() at refresh instead of the current RefreshToken(attrs["refresh"]) (create a new token from scratch instead of copying the claims from the previous one). That would require the query, though.

Reply:
The suggested change of user_id to user_pk doesn't pass tests, and I can't find any hint that that syntax would be supported by the Django ORM.

Reply:
However after looking at this a bit more, I don't think anything guarantees that api_settings.USER_ID_FIELD == User.id. So perhaps the query at TokenRefreshSerializer.validate() is unavoidable?